### ISSUE TITLE
Standard Unfolded Band Structure Plot

### ISSUE THREAD

**Issue by NewMaterialsLab** [SMTG-Bham/easyunfold#53]
I am having troubles plotting the Unfolded band structure for a disordered Heusler alloy using a meta-GGA functional. I follow the steps of the tutorial but the final plot move all the symmetry labels to the right. It looks that there is some problem distributing the data uniformly in the plot. I attach the png file I get. 
![unfold](https://github.com/user-attachments/assets/56bb4a95-24c3-4b9f-93b3-b308b00cfa5c)

**Comment by NewMaterialsLab** [comment]
Dear Sean,

Thanks for your fast response. I attach the files you asked here. I create the primitive cells path using sumo. Please, let me know if you need another file. 

Best regards,

Jose

[Debug.tgz](https://github.com/user-attachments/files/17639368/Debug.tgz)

**Comment by kavanase** [comment]
Thanks for sending the data.

I notice that you're using zero-weighted KPOINTS here (with the `--scf-kpoints` option), which we usually only do with hybrid DFT calculations (but these are meta-GGA calcs). Is there a reason for this? If you try doing the calculation with the typical GGA workflow (with `ICHARG = 11` and not zero-weighted kpoints etc), does it fix this issue?

I see zero weighted kpoints were used for the primitive cell calc and it seemed to work fine, but I'm not sure if this was because a previous `WAVECAR` was already present?

**Comment by NewMaterialsLab** [comment]
Hi Sean,

Thanks for your response. I am using r2SCAN functional. For meta-GGA functionals you cannot use ICHARG =11, calculation crashes. The procedure for meta-GGA functionals is the same than the one used for hybrid functional. I share VASP link:

https://www.vasp.at/wiki/index.php/Band-structure_calculation_using_meta-GGA_functionals

I used the same procedure for pc and sc. 

1) Optimization
2) Single point calculation saving WAVECAR
3) Band structure calculation using the IBZKPT file from step 2 and restarting using WAVECAR obtained in step 2.  The KPOINTS file used in step 3 for the pc was created with sumo and the KPOINTS file in step 3 for the supercell was created with easyfold.

As you pointed out, I plotted the pc band structure and everything looks fine.  I also checked the KPOINTS created be easyfold first recovering the IBZKPT and then appending the path and it looks fine to me too.

Thanks so much for your help.

Best regards,

Jose

**Comment by NewMaterialsLab** [comment]
Hi Sean,

I found the problem. In STEp 2, I was using as a reference KPOINTS file of the primitive cell I was using for the hybrid calculation. I think that is not clear enough in the documentation. It does not matter if you are doing GGA or Hybryd/metaGGA calculations, the KPOINTS reference fro the primitive cell should only contain the path you want to plot. 

Best regards,

Jose

**Comment by kavanase** [comment]
Hi Jose,
Thank you very much for looking into this and finding the issue!
I'll check the issue with the transformation matrix warning here, and update the docs to make this point about the primitive cell KPOINTS clearer.
Thanks again!
Seán

**Comment by kavanase** [comment]
Hi Jose,
I agree this was not clear in the docs. I have now updated so that `easyunfold` automatically handles the input kpoints appropriately (taking only the zero-weighted ones if mixed weighted + zero-weighted kpoints are present, otherwise taking all kpoints (assuming a LDA/GGA VASP band structure calculation)). 5c4434083675af965bdcc9d8197052ec80e19df1
I have also fixed the transformation matrix issue; it was using relative tolerances for each element in the matrix, so elements with values 1e-8 would then cause this to flag a 5% relative difference to 0 for example. Fixed now (df278f8f3af8f01ada366b34a8b501aedc22d91d)
